rules/xcodeproj.bzl

```diff
@@ -269,9 +278,22 @@ def _xcodeproj_impl(ctx):
             framework_search_paths.append("\"%s\"" % fi)
         target_settings["FRAMEWORK_SEARCH_PATHS"] = " ".join(framework_search_paths)
         target_settings["GCC_PREPROCESSOR_DEFINITIONS"] = " ".join(["\"%s\"" % d for d in target_info.cc_defines.to_list()])
+
         if target_info.product_type == "application":
             target_settings["INFOPLIST_FILE"] = "$BAZEL_STUBS_DIR/Info-stub.plist"
             target_settings["PRODUCT_BUNDLE_IDENTIFIER"] = target_info.bundle_id
+        else:
+            defines_without_equal_sign = []
```

Comment:
Based upon our discussion with @segiddins  this morning, I *think* we need to also add the `DEBUG` value to the list of defines for the debug xcconfig. This is done by cocoapods (which adds this build setting to xcode) and also by the bazel toolchain when it performs builds without xcode ... but our solution needs to do it as well I think.

Reply:
i see will add it here! wonder whether we can tell which config we are on at this point (passing DEBUG to release config can be a bad thing no?)

Reply:
this is a bunch of logic here -- could you extract it into a separate function?

Reply:
We should pass `DEBUG` for debug config, see https://github.com/yonaskolb/XcodeGen/blob/master/Docs/ProjectSpec.md#settings for docs on how to do per-config settings

Reply:
I still think this would be easier to follow if the logic for transforming defines for swift was extracted to a separate function

Reply:
we now pass `DEBUG` for debug config in the project level, all targets will inherit it from the project setting